### ISSUE TITLE
numpy.ma.expand_dims treats ma.masked differently from other singletons

### ISSUE THREAD

**Issue by christine-e-smit** [numpy/numpy#7424]
The numpy.ma.expand_dims returns numpy.ma.masked rather than an array if the input is numpy.ma.masked.

```
In [1]: import numpy.ma as ma

In [2]: ma.expand_dims(1,1)
Out[2]: array([1])

In [3]: ma.expand_dims(ma.masked,1)
Out[3]: masked
```

I am using numpy library version 1.10.1.

**Comment by sethtroisi** [comment]
Seems fixed at head
```
>>> import numpy.ma as ma
>>> ma.expand_dims(1,1)
array([1])
>>> ma.expand_dims(ma.masked,1)
masked_array(data=[--],
             mask=[ True],
       fill_value=1e+20,
            dtype=float64)
```

**Comment by christine-e-smit** [comment]
Looks good to me!

**Comment by christine-e-smit** [comment]
Actually, it appears to be fixed in version 1.16.4, which I've got installed on my machine.
